client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCacheManager.java

```diff
@@ -650,14 +648,15 @@ private Properties loadFromStream(InputStream stream) {
             RemoteCacheImpl<K, V> result = createRemoteCache(cacheName);
             RemoteCacheHolder rcc = new RemoteCacheHolder(result, forceReturnValueOverride == null ? configuration.forceReturnValues() : forceReturnValueOverride);
             startRemoteCache(rcc);
-            if (configuration.pingOnStartup()) {
-               // If ping not successful assume that the cache does not exist
-               // Default cache is always started, so don't do for it
-               if (!cacheName.equals(RemoteCacheManager.DEFAULT_CACHE_NAME) &&
-                     ping(result) == PingResult.CACHE_DOES_NOT_EXIST) {
-                  return null;
-               }
+
+            // If ping not successful assume that the cache does not exist
+            // Default cache is always started, so don't do for it
+            if (!cacheName.equals(RemoteCacheManager.DEFAULT_CACHE_NAME) &&
+                  ping(result) == PingResult.CACHE_DOES_NOT_EXIST) {
+               return null;
             }
+
+            result.resolveCompatibility();
```

Comment:
You are still pinging twice. Can we collapse the cache resolution with the compatibility check in a single ping ?